features/yield/FarmingCalculator.tsx

```diff
@@ -0,0 +1,172 @@
+import { Box, TextField, Typography, Grid } from "@material-ui/core";
+
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import EmpState from "../../containers/EmpState";
+import Balancer from "../../containers/Balancer";
+
+import { getUmaPrice } from "./UmaTokenPrice";
+
+const FormInput = styled.div`
+  margin-top: 20px;
+  margin-bottom: 20px;
+`;
+
+const WEEKS_PER_YEAR = 52;
+
+const YieldCalculator = () => {
+  const { empState } = EmpState.useContainer();
+  const { usdPrice, pool } = Balancer.useContainer();
+  const [umaPrice, setUmaPrice] = useState<number | null>(4.0);
+  const [yUSDAdded, setyUSDAdded] = useState<string | null>("1000");
```

Comment:
Also, FYI we can easily query the user's shares and % of total shares in pool if they connect their web3 provider. Maybe we should make farming calculator only available if they connect their web3 provider and then we can just detect their shares amount.

Try this query:
```
{
pools(where: {tokensList: ["0x81ab848898b5ffd3354dbbefb333d5d183eedcb5", "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"]}) {
    shares(first:1000) {
      userAddress {
        id
      }
      balance
    }
  }
}
```

@ https://thegraph.com/explorer/subgraph/balancer-labs/balancer

Reply:
Ye, This is the query that I used in the LP mining script. might be nice to update the table with their actual amount provided in the same way we update the token price in the yUSD payout calculator. however I think this should perhaps be a separate PR as this is already 300+ lines changed.

Reply:
Yeah I think we should think of this PR as setting up the TSX components/container

Reply:
cool. will leave for a leter PR if we deem it wanted